app/src/main/java/com/kickstarter/viewmodels/CommentsViewModel.java

```diff
@@ -200,6 +185,58 @@ public CommentsViewModel(final @NonNull Environment environment) {
       .take(1)
       .compose(bindToLifecycle())
       .subscribe(__ -> refresh.onNext(null));
+
+    final Observable<Update> update = projectOrUpdate.map(Either::right);
+
+    // Todo: add a pageCount to RecyclerViewPaginator to track loading newer comments.
+    Observable.combineLatest(project, update, Pair::create)
+      .compose(takeWhen(nextPage.skip(1)))
```

Comment:
🤔 this is what I assumed (admittedly I didn't change the tracking logic itself when refactoring here). funny enough `nextPage` does not emit immediately on initial load, but it does emit when there are no comments. will look into `RecyclerViewPaginator` to check out this logic